### DISCUSSION TITLE
Neptune 2/2S v1.3 source code

### DISCUSSION THREAD

**Discussion by TheFeralEngineer** [NARUTOfzr/ZNP-Robin-Nano-V1.2#6] [General]
can you post the source code for the robin v1.3 or will the neptune 3 source code work with a different pin file?

**Comment by NARUTOfzr** [comment]
The firmware for the Neptune2 series is provided by makerbase and the source code is not available to us. But you can try using the official marlin2. See the documentation for how to configure it:
https://github.com/NARUTOfzr/ZNP-Robin-Nano-V1.2-V1.3/blob/master/1-Function%20extension/Neptune%202%20series%20marlin%20configuration.docx

**Comment by TheFeralEngineer** [comment]
Sorry about the weird formatting, I was replying to the emails, not on here. 

Does Marlin 2.1.2 work on the ZNP Robin Nano V2.1 for the neptune 3? Would a TFT43 or MKS H43 work in place of the original Dwin T5l0 screen? I noticed that the marlin build says it's for a neptune 3 when I do an EEPROM load into octoprint.

Also, will marlin 2.1.2 work on the neptune x with the v1.3 usbmod firmware?

**Comment by NARUTOfzr** [comment]
About the compatibility between the motherboard and the screen. Maybe you can ask the maker base technician directly. Because this is a customized motherboard from mks.
I haven't tried these screens and can't give you a definite answer, but I will ask the mks technician about your questions tomorrow.
About the marlin2 configuration method of neptune X, we still need to try.




---Original---
From: ***@***.***&gt;
Date: Wed, Dec 21, 2022 08:20 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [NARUTOfzr/ZNP-Robin-Nano-V1.2-V1.3] Neptune 2/2S v1.3 sourcecode (Discussion #6)




 
Sorry about the weird formatting, I was replying to the emails, not on here.
 
Does Marlin 2.1.2 work on the ZNP Robin Nano V2.1 for the neptune 3? Would a TFT43 or MKS H43 work in place of the original Dwin T5l0 screen? I noticed that the marlin build says it's for a neptune 3 when I do an EEPROM load into octoprint.
 
Also, will marlin 2.1.2 work on the neptune x with the v1.3 usbmod firmware?

**Reply by TheFeralEngineer** [reply]
Thank you for the reply. Please ask MKS for me. They never answer my questions directly :(. I'm currently running Marlin 2.1.2 on my neptune 2. Thank you :). 

If I tried to flash marlin 2.1.2 into my neptune 3, what would I need to do to make it work? I know there are a lot of modifications in 2.0.9.3.1, but I'm not sure how many are necessary if I didn't use the Dwin screen.

**Comment by NARUTOfzr** [comment]
Hi!@TheFeralEngineer 
 Here's the bad news:
I asked the mks technicians and they said that the motherboard is compatible with the MKS_H43 screen, but only with the makerbase-GitHub firmware version.https://github.com/makerbase-mks/Mks-Robin-Nano-Marlin2.0-Firmware   .There is no information about this motherboard in marlin2.1.2.

But there are two pieces of good news:
1. I see that David-McKenna has ported the information from that motherboard to marlin2.1.1, you might want to try that marlin version. I think it will also be compatible with mks_H43.https://github.com/NARUTOfzr/Neptune_3/issues/22#issuecomment-1288105793

2, our software engineer says: Neptune3Pro firmware version marlin2.1.1 is compatible with using MKS-H43 screen. The open source files are currently being sorted out, and maybe the source code will be shared with you next week.

A note of caution: I'm not sure if the motherboard RJ11 wiring port will work with the MKS_H43 screen, you need to check the wiring order to prevent burning the screen or motherboard. You can also try this method of wiring:https://github.com/makerbase-mks/MKS-H43/blob/main/Images/H43_ROBIN_NANO_V1_V2.jpg

Finally! Merry Christmas.

**Reply by TheFeralEngineer** [reply]
Thank you and Merry Christmas to you as well! 

Thank you for this information. It is very helpful. My only reason for wanting to use the H43 screen is because the Makerbase DGUS HMI project file loads on my computer without any region issues. I will check David's firmware and see if it works well.  I am under the impression that the Neptune 3 has been discontinued and I would like to be able to further the development of the user HMI, either in the form of the Dwin T5L0 screen or the MKS H43 for users that do not wish to change to Klipper firmware and would like more functionality from the HMI. The Neptune 3 screen is missing a lot of features and I would like to add them. 

I purchased an H43 screen a while ago and returned it because I couldn't get it to respond via the RJ11 port, as you mentioned. I was able to power the screen with the serial UART pins next to the wifi socket, but I don't believe I enabled serial 2 in the firmware to make it work and returned it because I was frustrated 😂🙄. I'm currently using those serial UART pins to connect to an MKS Pi and run my Neptune 3 with Octoprint, but I still use the Dwin screen for basic movements. If the Neptune 3 Pro screen were a plug and play replacement for the Neptune 3 screen, I believe a lot of Neptune 3 owners would want to buy them. 

How much trouble would it be to port the Neptune 3 Pro v2.1.1 firmware to the Neptune 3 standard? Are they both running the same STM32F4 chip? If the H43 will work on that firmware and that firmware would load onto the V2.1 board, that would be a great mod/upgrade for the Neptune 3. I know that Pro V2.2 motherboard does not contain the pins for a BLTouch, but what other differences are there between V2.1 and V2.2 boards? 

For the future, please let your product developers know that there are a lot of people like to modify their machines and removing the BLTouch pins gives them one less opportunity to customize their machine if they do not like the inductive ABL sensor. My Neptune 3 isn't running the strain gauge anymore. I have a BLTouch, dual independent Z axis motors (using the E1 driver socket), direct drive extruder and linear rails and my Neptune 2 has the original Neptune 3 metal extruder, PEI sheet, BLTouch and Marlin 2.1.2 with more modifications to come. I really like machines that I can tinker with 😊

Thank you and Merry Christmas once again 😋

**Comment by NARUTOfzr** [comment]
Thank you! 😋

Neptune3 is currently out of production, but it is unclear whether subsequent plans will improve it or keep it in production. At present, there is no specialized software developer for maintenance of the firmware of Neptune3. Most of the time, I carried out some small improvements, but my time is limited. With the new machines coming out, I have even less time.

Neptune3&Neptune3Pro used STM32F4 chips. V2.1 motherboards and V2.2 motherboards were almost universal, but some pin layouts were not quite the same. The screen of Neptune3Pro can be used in V2.1 motherboard, but it needs to be used with the motherboard firmware of Neptune3Pro (because the screen needs to interact with the motherboard firmware). It is very troublesome to transplant the code, there may be various bugs, we need to be very patient to check and repair, otherwise we also hope to update the marlin version according to the needs of users.

Thank you for your advice. I will feed it back to our R&D team. Fun is our continuous motivation, come on! 😊

**Comment by TheFeralEngineer** [comment]
Thank you for this information. I ordered another H43 to test from aliexpress ($25 vs $50 on amazon 🤯), so I plan on moving forward with some tests of Marlin 2.1.1 with that screen to see if I can make it work. If it is possible, I will probably attempt to port it to Marlin 2.1.2 and modify the DGUS project to be more user friendly to give others the option of using this board for their Neptune 3. I wouldn't mind trying to use the Neptune 3 Pro firmware on my Neptune 3 if I could get a pro screen to test😋. Do you know if the DGUS project for the Neptune 3 pro would load on the H43? 

I think that a Neptune 3v2 with bowden tube, single Z screw and new DUGS screen would be a welcome addition to the Neptune lineup as some people do not prefer direct dive extruders. If Elegoo plans to phase out the Neptune 2 and 2s, a Neptune 3v2 might be a perfect replacement in that price point. Ultimately, I will probably switch to Klipper, but I want to help those in the community that wish to continue using Marlin and will update my progress on my youtube channel. I have a little experience with C++ and HMI development with Nextion HMI software and would like something to work on to help me develop my skills further. The DGUS stuff is a little more difficult, but learning is my favorite hobby.

**Reply by NARUTOfzr** [reply]
Sorry for the late reply!
Nope.DGUS project is used on the DWIN screen. Neptune3Pro uses a Nextion screen.
The source code of the motherboard firmware has been uploaded and you can download it here:https://github.com/NARUTOfzr/Neptune_3_Pro

The source code for the screen project is still in preparation and will be uploaded to this link soon.

**Reply by TheFeralEngineer** [reply]
That's very interesting... I have a nextion screen on hand and I know the nextion HMI software 🤔🤔. Thank you 🙂
